snuba/utils/streams/consumers/backends/synchronized.py

```diff
@@ -0,0 +1,271 @@
+from abc import ABC, abstractmethod
+from collections import defaultdict
+from dataclasses import dataclass, field
+from threading import Event, Lock
+from typing import Callable, Generic, Mapping, MutableMapping, Optional, Set, Sequence
+
+from snuba.utils.concurrent import execute
+from snuba.utils.streams.consumers.backends.abstract import ConsumerBackend
+from snuba.utils.streams.consumers.types import Message, TOffset, TStream, TValue
+
+
+ConsumerGroup = str
```

Comment:
This is just for readability (it makes some of the annotations below a bit more self-documenting.)